### PR TITLE
ci: update ubuntu:20.04 builders to 22.04

### PR DESCRIPTION
**Pull request by cuviper**
This is mostly just maintenance to avoid bitrotting, but 22.04 also updates to cmake 3.22, so they don't need the manual builds from #113714 anymore.

### CODE REVIEW

File: src/ci/docker/host-x86_64/dist-various-1/Dockerfile [hunk 1]
```diff
@@ -73,8 +74,8 @@ RUN env \
     CXX=arm-linux-gnueabihf-g++ CXXFLAGS="-march=armv6 -marm -mfpu=vfp" \
     bash musl.sh armhf && \
     env \
-    CC=arm-linux-gnueabihf-gcc CFLAGS="-march=armv7-a" \
-    CXX=arm-linux-gnueabihf-g++ CXXFLAGS="-march=armv7-a" \
+    CC=arm-linux-gnueabihf-gcc CFLAGS="-march=armv7-a -mfpu=vfp3" \
+    CXX=arm-linux-gnueabihf-g++ CXXFLAGS="-march=armv7-a -mfpu=vfp3" \
```
Comment (wesleywiser) [inline src/ci/docker/host-x86_64/dist-various-1/Dockerfile:78]: Just curious why this change was needed. Did gcc change the default here or something?

Comment (cuviper) [reply src/ci/docker/host-x86_64/dist-various-1/Dockerfile:78]: I'm not exactly sure what changed, but I do see that a lot of GCC tests explicitly specify this combination. Without it, the musl build fails:

```
+ cd musl-1.2.3
+ ./configure --enable-debug --disable-shared --prefix=/musl-armv7hf
checking for C compiler... arm-linux-gnueabihf-gcc
checking whether C compiler works... no; compiler output follows:
cc1: error: '-mfloat-abi=hard': selected architecture lacks an FPU
```

Comment (cuviper) [reply src/ci/docker/host-x86_64/dist-various-1/Dockerfile:78]: This may be a Ubuntu change -- in 20.04, `arm-linux-gnueabihf-gcc -v` reported configuration `--with-arch=armv7-a --with-fpu=vfpv3-d16`, and in 22.04 that's `--with-arch=armv7-a+fp` (w/o separate fpu).

Comment (cuviper) [reply src/ci/docker/host-x86_64/dist-various-1/Dockerfile:78]: In `man gcc` it says under armv7-a, "+fp The VFPv3 floating-point instructions, with 16 double-precision registers.  The extension +vfpv3-d16 can be used as an alias for this extension." So we should probably just match that and it will be consistent with what we had before.

**Comment by Mark-Simulacrum** [comment]
@bors r+ rollup=never

Nothing here looks too suspicious...

**Comment by Kobzol** [comment]
I *think* that the bump from 20.04 to 22.04 updates the glibc version. Do we care about this for some of the modified dist builders? Or is this only relevant for x64 Linux gnu?

**Comment by cuviper** [comment]
Hmm, I suppose it is relevant to any `*-linux-gnu` that uses the Ubuntu cross compilers, but it doesn't matter as much for those that don't provide host tools. So `dist-powerpc64le-linux` is a host but not using the Ubuntu toolchain; `dist-various-1` and `-2` have a few `linux-gnu` that are non-host, using Ubuntu's toolchain.

Do we care about `libstd.so` glibc versions on those? (`libstd.rlib` should be fine.)

**Comment by cuviper** [comment]
If we do care, maybe we could fork those `linux-gnu` targets into a `dist-various-3` or something, that explicitly holds back on `ubuntu:20.04`. Long term, it would be better to use crosstool-ng toolchains when we want to control glibc versions.

**Comment by Mark-Simulacrum** [comment]
Can you check against https://doc.rust-lang.org/beta/rustc/platform-support.html? I think if we don't document a version we should be fine to bump as a nonevent, otherwise let's bump the documentation at least. (My quick scroll suggests it's not unlikely this doesn't change anything, but I'm not sure)

**Comment by matthiaskrgr** [comment]
Hmm, `dist-x86_64-musl` is running for 2.5 hours already, hopefully that is just a one-off :sweat:

**Comment by cuviper** [comment]
> Can you check against https://doc.rust-lang.org/beta/rustc/platform-support.html? I think if we don't document a version we should be fine to bump as a nonevent, otherwise let's bump the documentation at least.

The glibc versions under Tier 2 (without host) should be updated, but they're already wrong. They should have been updated for #81521 at least, and there's a deja vu comment about compat and dynamic linking... :sweat_smile: 

I'll send an update, including a footnote about that applicability.

**Comment by cuviper** [comment]
> Hmm, `dist-x86_64-musl` is running for 2.5 hours already, hopefully that is just a one-off 😓

LLVM alone took over 1.5 hours in that build, but only 5 minutes in the next -- hooray for sccache!
